### PR TITLE
Media airing status

### PR DESCRIPTION
**Pull request by matthewdias**

### CODE REVIEW

File: app/models/concerns/media.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module Media
   extend ActiveSupport::Concern
 
+  STATUSES = %w[nya upcoming current finished].freeze
```
Comment (NuckChorris) [inline app/models/concerns/media.rb]: n-nyaa?  What's this, nya?

File: app/models/concerns/media.rb [hunk 2]
```diff
@@ -45,6 +47,17 @@ module Media
...
+    scope :past, -> { where('start_date <= ?', Date.today) }
+    scope :finished, -> { past.where('end_date < ?', Date.today) }
+    scope :current, -> do
+      past.where('end_date >= ? OR end_date IS ?', Date.today, nil)
+    end
+    scope :future, -> { where('start_date > ?', Date.today) }
+    scope :upcoming, -> do
+      future.where('start_date <= ?', Date.today + 3.months)
+    end
+    scope :nya, -> { future.where('start_date > ?', Date.today + 3.months) }
```
Comment (NuckChorris) [inline app/models/concerns/media.rb]: And here?

File: app/models/concerns/media.rb [hunk 3]
```diff
@@ -45,6 +47,17 @@ module Media
...
     delegate :year, to: :start_date, allow_nil: true
 
+    scope :past, -> { where('start_date <= ?', Date.today) }
+    scope :finished, -> { past.where('end_date < ?', Date.today) }
+    scope :current, -> do
+      past.where('end_date >= ? OR end_date IS ?', Date.today, nil)
+    end
+    scope :future, -> { where('start_date > ?', Date.today) }
+    scope :upcoming, -> do
+      future.where('start_date <= ?', Date.today + 3.months)
```
Comment (NuckChorris) [inline app/models/concerns/media.rb:62]: What's the logic here?

File: app/models/concerns/media.rb (hunk 1 above)
Comment (toyhammered) [reply app/models/concerns/media.rb]: nyaa nyaa!

Comment (toyhammered) [reply app/models/concerns/media.rb]: not yet airing maybe?

Comment (matthewdias) [reply app/models/concerns/media.rb]: Yeah not yet airing. I didn't know what to call it so just went with what the client is already using. Should I change it?

File: app/models/concerns/media.rb (hunk 3 above)
Comment (matthewdias) [reply app/models/concerns/media.rb:62]: Upcoming = it starts in the next 3 months.

File: app/models/concerns/media.rb (hunk 2 above)
Comment (matthewdias) [reply app/models/concerns/media.rb]: Not yet aired = it starts in the future outside of the upcoming range.

Comment (matthewdias) [reply app/models/concerns/media.rb]: Could maybe change this to include all future for queries, but I did it like this to keep consistency with the attribute.

Comment (crse) [reply app/models/concerns/media.rb]: Do we need to be announced for anime that does not have fixed date yet?

Comment (matthewdias) [reply app/models/concerns/media.rb]: could you rephrase that? im not sure what you're asking.

Comment (matthewdias) [reply app/models/concerns/media.rb]: nevermind, I gotcha. yeah that would be a good addition, ill get that in.

Comment (matthewdias) [reply app/models/concerns/media.rb]: or should i just roll that into "not yet aired"? cc @NuckChorris

File: app/models/concerns/media.rb (hunk 1 above)
Comment (NuckChorris) [reply app/models/concerns/media.rb]: I think `not_yet_airing` would be nicer, there's not a lot of reason to save those like, 10 bytes
